app/validators/proxy_features_validator.rb

```diff
@@ -0,0 +1,15 @@
+class ProxyFeaturesValidator < ActiveModel::Validator
```

Comment:
Should probably be an EachValidator, then there's no need to iterate over attributes yourself.